cmd/backup/script.go

```diff
@@ -421,21 +500,55 @@ func (s *script) stopContainers() (func() error, error) {
 			}
 		}
 
-		if len(restartErrors) != 0 {
+		var scaleUpErrors []error
+		if isDockerSwarm {
+			for _, service := range servicesToScaleDown {
+				updatedService, _, err := s.cli.ServiceInspectWithRaw(context.Background(), service.ID, types.ServiceInspectOptions{})
+				if err != nil {
+					scaleUpErrors = append(scaleUpErrors, err)
+					continue
+				}
+
+				if updatedService.PreviousSpec == nil {
+					scaleUpErrors = append(
+						scaleUpErrors,
+						errors.New("(*script).stopContainersAndServices: service does not have PreviousSpec, cannot scale back up."),
+					)
+					continue
+				}
+
+				updatedService.Spec.Mode.Replicated.Replicas = updatedService.PreviousSpec.Mode.Replicated.Replicas
```

Comment:
as mentioned above, not sure if backup can rely on PreviousSpec. Would also be a possibility to store the Replica count in a variable and update to that. ( Or maybe a deep copy of the original spec and that can be compared to before trying to scale up. If spec what is expected does not match previousspec, there was somebody doing something in the background?

Reply:
>  there was somebody doing something in the background?

Yeah, that's a valid concern. I'd probably do it like you describe and wrap the service object like

```go
type handledService {
    serviceID                                       string
    replicaCountBeforeScaleDown uint64
}
```

Like this, the script can lookup the current spec at scale up time and only change the replica count to what it was before scaling down. This effectively allows users to change everything _but_ the replica count while a backup is performed. I have a hard time thinking of an approach that allows updating the desired replica count while a backup is happening. But I guess this is ok, as long as it's mentioned in the docs.